cmd/microcloud/main_init.go

```diff
@@ -757,6 +766,20 @@ func (c *initConfig) setupCluster(s *service.Handler) error {
 				}
 			}
 
+			if s.Type() == types.MicroOVN {
+				microOvnBootstrapConf := make(map[string]string)
+				for _, system := range c.systems {
+					// If the system needs to be bootstrapped with a custom ovn encapsulation ip for geneve, add it to the init config.
+					if system.OVNGeneveAddr != "" {
+						microOvnBootstrapConf["ovn-encap-ip"] = system.OVNGeneveAddr
+					}
+				}
```

Comment:
Maybe the test failures have to do with this?

Looks like you're iterating over the `c.systems` map and selecting an effectively random address for the geneve address. I think you might want to grab the config from `bootstrapSystem` like with `MicroCephInternalNetworksubnet`.

Reply:
fixed